src/clr-angular/forms/datepicker/date-container.ts

```diff
@@ -28,35 +42,107 @@ import { LocaleHelperService } from './providers/locale-helper.service';
             <clr-icon shape="calendar" class="datepicker-trigger-icon"></clr-icon>
         </button>
         <clr-datepicker-view-manager *clrIfOpen clrFocusTrap></clr-datepicker-view-manager>
+    </ng-template>
+    
+    <ng-template #newLayout>
+      <ng-content select="label"></ng-content>
+      <div class="clr-control-container" [ngClass]="controlClass()">
+        <div class="clr-input-wrapper">
+          <div class="clr-input-group" [class.clr-focus]="focus">
+            <ng-container *ngTemplateOutlet="clrDate"></ng-container>
+            <button type="button" class="datepicker-trigger" (click)="toggleDatepicker($event)" *ngIf="isEnabled">
+              <clr-icon shape="calendar" class="clr-input-group-icon-action"></clr-icon>
+            </button>
+            <clr-datepicker-view-manager *clrIfOpen clrFocusTrap></clr-datepicker-view-manager>
+          </div>
+          <clr-icon class="clr-validate-icon" shape="exclamation-circle"></clr-icon>
+        </div>
+        <ng-content select="clr-control-helper" *ngIf="!invalid"></ng-content>
+        <ng-content select="clr-control-error" *ngIf="invalid"></ng-content>
+      </div>
+    </ng-template>
+    
+    <ng-template #clrDate>
+      <ng-content select="[clrDate]"></ng-content>
+    </ng-template>
+    
+    <ng-container *ngIf="newFormsLayout; then newLayout else oldLayout"></ng-container>
     `,
   providers: [
     ControlIdService,
     IfOpenService,
     LocaleHelperService,
+    IfErrorService,
+    ControlClassService,
+    FocusService,
+    NgControlService,
     DateIOService,
     DateNavigationService,
     DatepickerEnabledService,
     DateFormControlService,
   ],
-  host: { '[class.date-container]': 'true' },
+  host: {
+    '[class.date-container]': '!newFormsLayout',
+    '[class.clr-form-control]': 'newFormsLayout',
+  },
 })
 export class ClrDateContainer implements DynamicWrapper, OnDestroy {
-  // Unused but have to add it :-(
   _dynamic: boolean = false;
+  invalid = false;
+  focus = false;
 
-  private _sub: Subscription;
+  private subscriptions: Subscription[] = [];
 
   constructor(
     private _ifOpenService: IfOpenService,
     private _dateNavigationService: DateNavigationService,
     private _datepickerEnabledService: DatepickerEnabledService,
-    private dateFormControlService: DateFormControlService
+    private dateFormControlService: DateFormControlService,
+    private ifErrorService: IfErrorService,
+    private focusService: FocusService,
+    private controlClassService: ControlClassService,
+    @Optional() private layoutService: LayoutService,
+    @Optional()
+    @Inject('IS_NEW_FORMS_LAYOUT')
```

Comment:
I feel bad requiring this because it seems like such a small thing, but typing is 👑: can you please switch to an `InjectionToken<boolean>` rather than a plain string, so we get proper typing when injecting this?